store/src/java/com/zimbra/cs/event/SolrEventStore.java

```diff
@@ -58,6 +77,158 @@ protected void deleteEventsByDataSource(String dataSourceId) throws ServiceExcep
         solrHelper.execute(accountId, req);
     }
 
+    @Override
+    public Long getContactFrequencyCount(String contact, ContactAnalytics.ContactFrequencyEventType eventType, ContactAnalytics.ContactFrequencyTimeRange timeRange) throws ServiceException {
+        SolrQuery solrQuery = new SolrQuery();
+        solrQuery.setRows(0);
+        solrQuery.setQuery(getContactFrequencyQueryForTimeRange(timeRange));
+        solrQuery.addFilterQuery(getQueryToSearchContact(contact, eventType));
+
+        if (solrHelper.needsAccountFilter()) {
+            solrQuery.addFilterQuery(getAccountFilter(accountId));
+        }
+
+        QueryResponse response = (QueryResponse) solrHelper.executeRequest(accountId, solrQuery);
+        return response.getResults().getNumFound();
+    }
+
+    private String getQueryToSearchContact(String contact, ContactAnalytics.ContactFrequencyEventType eventType) {
+        switch (eventType) {
+            case SENT:
+                return searchContactByEventType(contact, Event.EventType.SENT, Event.EventContextField.RECEIVER).toString();
+            case RECEIVED:
+                return searchContactByEventType(contact, Event.EventType.RECEIVED, Event.EventContextField.SENDER).toString();
+            default:
+                return getQueryToSearchContactAsSenderOrReceiver(contact).toString();
+        }
+    }
+
+    private String getContactFrequencyQueryForTimeRange(ContactAnalytics.ContactFrequencyTimeRange timeRange) throws ServiceException {
+        if(ContactAnalytics.ContactFrequencyTimeRange.FOREVER.equals(timeRange)) {
+            return new MatchAllDocsQuery().toString();
+        }
+        TermRangeQuery rangeQuery = TermRangeQuery.newStringRange(LuceneFields.L_EVENT_TIME, getContactFrequencyQueryStartDate(timeRange), "NOW", true, true);
+        return rangeQuery.toString();
+    }
+
+    private String getContactFrequencyQueryStartDate(ContactAnalytics.ContactFrequencyTimeRange timeRange) throws ServiceException {
+        switch (timeRange) {
+            case LAST_DAY:
+                return "NOW-1DAY";
+            case LAST_WEEK:
+                return "NOW-7DAY";
+            case LAST_MONTH:
+                return "NOW-1MONTH";
+            default:
+                throw ServiceException.FAILURE("Time range not supported " + timeRange, new NotImplementedException());
```

Comment:
You can probably use `ServiceException.INVALID_REQUEST` here instead of NotImplementedException.